lightglue/utils.py

```diff
@@ -59,26 +119,15 @@ def load_image(path: Path, grayscale: bool = False, resize: int = None,
     return numpy_image_to_torch(img), torch.Tensor(scales)
 
 
-def match_pair(extractor, matcher, image0, image1, scales0=None, scales1=None, device=None):
-    data = {'image0': image0[None], 'image1': image1[None]}
-    img0, img1 = data['image0'], data['image1']
-    feats0, feats1 = extractor({'image': img0}), extractor({'image': img1})
-    pred = {**{k+'0': v for k, v in feats0.items()},
-            **{k+'1': v for k, v in feats1.items()},
-            **data}
-    pred = {**pred, **matcher(pred)}
-    pred = {k: v.to(device).detach()[0] if
-            isinstance(v, torch.Tensor) else v for k, v in pred.items()}
-    if scales0 is not None:
-        pred['keypoints0'] = (pred['keypoints0'] + 0.5) / scales0[None] - 0.5
-    if scales1 is not None:
-        pred['keypoints1'] = (pred['keypoints1'] + 0.5) / scales1[None] - 0.5
-    del feats0, feats1
-    if 'cuda' in str(device):
-        torch.cuda.empty_cache()
-
-    # create match indices
-    matches0, mscores0 = pred['matches0'], pred['matching_scores0']
-    valid = matches0 > -1
-    matches = torch.stack([torch.where(valid)[0], matches0[valid]], -1)
-    return {**pred, 'matches': matches, 'matching_scores': mscores0[valid]}
+def match_pair(extractor, matcher,
+               image0: torch.Tensor, image1: torch.Tensor,
+               resize: Optional[int] = 1024,
```

Comment:
Why force the resizing to 1024? Just add a `**kwargs` and forward to `extract`

Reply:
changed to kwargs